Automattic/jetpack
# AI Usage bar: change color for over soft limit

When highest tier is over limit we treat it as a soft limit, yet the color is the same as when it requires an upgrade.

Fixes #34319 

## Proposed changes:
This PR adds an extra prop to know if the site requires an upgrade and uses a yellowish color for the soft limit. The usual red for required upgrades remain.

<img width="292" alt="image" src="https://github.com/Automattic/jetpack/assets/157240/1cf83f17-0d1e-4914-86c3-dc1414ef4c62">


### Other information:


## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Go to the block editor of your test site and add an AI Assistant block
* Dispatch the following action to set the site as being over limit but not require an upgrade

```
wp.data.dispatch( 'wordpress-com/plans' ).storeAiAssistantFeature( {
    "hasFeature": true,
    "isOverLimit": true,
    "requestsCount": 20,
    "requestsLimit": 20,
    "requireUpgrade": false,
    "upgradeType": "default",
    "usagePeriod": {
        "currentStart": null,
        "nextStart": null,
        "requestsCount": 101
    },
    "currentTier": {
        "value": 100,
	"limit": 100,
    },
    "nextTier": {
        "slug": "jetpack_ai_yearly",
        "limit": 999999999,
        "value": 1,
        "readable-limit": "Unlimited"
    },
   "tierPlansEnabled": true,
} );
```

- see the usage panel on the side, it should be yellow

Change on the above payload `requireUpgrade` value to switch to different cases
